Use const char where needed. Taken from an upstream commit, as seen in
this GitHub mirror's commit:

https://github.com/x3ro/VisualBoyAdvance/commit/722a30419a30345be3c2ebf6ef85689067ea7005
--- src/GBA.cpp.orig	2004-05-13 09:37:59.000000000 -0500
+++ src/GBA.cpp	2018-10-07 22:14:50.000000000 -0500
@@ -1130,7 +1130,7 @@
 bool CPUIsZipFile(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".zip") == 0)
@@ -1145,7 +1145,7 @@
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -1169,7 +1169,7 @@
 bool CPUIsGBABios(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -1189,7 +1189,7 @@
 bool CPUIsELF(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
     
     if(p != NULL) {
       if(_stricmp(p, ".elf") == 0)
--- src/Util.cpp.orig	2004-05-20 12:42:37.000000000 -0500
+++ src/Util.cpp	2018-10-07 22:21:03.000000000 -0500
@@ -478,7 +478,7 @@
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -502,7 +502,7 @@
 bool utilIsGBImage(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gb") == 0)
@@ -522,7 +522,7 @@
 bool utilIsZipFile(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".zip") == 0)
@@ -537,7 +537,7 @@
 bool utilIsRarFile(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".rar") == 0)
@@ -552,7 +552,7 @@
 bool utilIsGzipFile(const char *file)
 {
   if(strlen(file) > 3) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gz") == 0)
--- src/gb/GB.cpp.orig	2004-05-13 14:35:26.000000000 -0500
+++ src/gb/GB.cpp	2018-10-07 22:17:32.000000000 -0500
@@ -567,7 +567,7 @@
 bool gbIsGameboyRom(char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gb") == 0)
